### PR TITLE
grpc: moving the last of the gRPC mux classes to factory based extension

### PR DESCRIPTION
**Pull request by alyssawilk**
200K savings for E-M, and cleaner directory structure.

Risk Level: medium (moving lots of code around)
Testing: n/a (existing tests)
Docs Changes: n/a
Release Notes: inline

### CODE REVIEW

File: source/server/server.cc [hunk 1]
```diff
@@ -955,18 +954,15 @@ void InstanceImpl::terminate() {
...
-    factory->shutdownAll();
-  }
-  factory =
-      Config::Utility::getFactoryByName<Config::MuxFactory>("envoy.config_mux.grpc_mux_factory");
-  if (factory) {
-    factory->shutdownAll();
+  std::vector<std::string> muxes = {
+      "envoy.config_mux.new_grpc_mux_factory", "envoy.config_mux.grpc_mux_factory",
+      "envoy.config_mux.delta_grpc_mux_factory", "envoy.config_mux.sotw_grpc_mux_factory"};
+  for (auto name : muxes) {
```
Comment (adisuissa) [inline source/server/server.cc:960]: nit:
```suggestion
  for (const auto& name : muxes) {
```

**Review by adisuissa — APPROVED** [review]
Impressive refactoring work, thanks!
Small nit but otherwise LGTM!

**Review by adisuissa — DISMISSED** [review]
LGTM, thanks!

**Comment by alyssawilk** [comment]
coverage flake was unrelated but I'm bumping down coverage just while I'm int here

**Review by RyanTheOptimist — APPROVED** [review]
<3

**Pull request merged**